buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+  `kotlin-dsl`
+}
```

Comment:
I think Square projects use (prefer?) the `apply plugin` syntax.
Same with Groovy build script style with kts style.  IIRC, there's still some perf issue related to using buildSrc.

How about mimicking the Gradle setup style of square/retrofit, square/wire, or cashapp/sqldelight?

Reply:
That's fair in those projects since they don't use kotlin gradle script and there's no difference to the user. Here they do matter because the modern `plugins {}` syntax will actually make kotlin gradle script generate the type-safe accessors for plugin extensions, whereas the `apply()` syntax won't. We do the same in kotlinpoet for this reason: https://github.com/square/kotlinpoet/blob/master/kotlinpoet/build.gradle.kts.

In general for kotlin gradle dsl you should only use the `apply(plugin = ...)` syntax when conditionally applying it (like the root `build.gradle.kts` file here does).

Reply:
Yep, but if we use the Groovy build script style instead (my second point), the apply vs plugins syntax becomes moot, I guess.  The referenced projects also use the `apply` syntax.  I think KotlinPoet is the only repo to use Kotlin DSL atm, fwiw.